website/cue/reference/remap/functions/map_keys.cue

```diff
@@ -0,0 +1,70 @@
+package metadata
+
+remap: functions: map_keys: {
+	category: "Enumerate"
+	description: #"""
+		Map the keys within an object.
+
+		If `recursive` is enabled, the function iterates into nested
+		objects, using the following rules:
```

Comment:
I realized we currently say [this](https://vector.dev/docs/reference/vrl/#lack-of-recursion) about recursion in our general VRL docs:

> VRL lacks recursion capabilities, making it impossible to create large or infinite loops that could stall VRL programs or needlessly drain memory.

I mean, _technically_ you can create large loops, for example, by getting bogus events with many levels of nesting.

We currently _don't_ account for that in VRL, but I'm also not sure if we should. Perhaps we want to add a recursion limit to Vector's `Value` type itself?

Luckily, if we do want to enforce a recursion limit, we can centralize this in the `ValueIter`, which is used by all three VRL iteration functions.

We could also allow people to define their own recursion limit (e.g. have `recursive: true` default to a sane limit, and allow `recursive: 3` to allow 3-level deep recursion. We'd limit the integer value to between `1` and `...` a sane higher limit.

Reply:
Maybe the doc could be updated to say we don't allow infinite recursion? Or would we, with the changes here?

Reply:
Well. We don't allow infinite recursion, but we _technically_ support a [128-level deep recursive object](https://github.com/serde-rs/json/blob/b899429dfb63d598f35f2f4c3cd96e0891472cbc/src/de.rs#L56), which would be slow to iterate over.

I think we're good to ship this as-is, as the 128-level recursion limit avoids blowing out the stack, and any further restrictions can be considered if we do see use-cases of people running into problems here.